### ISSUE TITLE
Java script executor fails to accept webelement when decorated with EventFiringDecorator

### ISSUE THREAD

**Issue by shroffsagar** [SeleniumHQ/selenium#9522]
## 🐛 Bug Report

When we register a WebDriverListener implementation with EventFiringDecorator the JavascriptExecutor.executeScript breaks and fails to accept the decorated web element passed in callback function. 

## To Reproduce

Add a class which implements WebDriverListener and override afterFindElement
```
@Override
    public void afterFindElement(WebDriver driver, By locator, WebElement result) {
        ((JavascriptExecutor)driver).executeScript("arguments[0].setAttribute('style','background: yellow; border: 2px solid red;')", result);
    }
```

Now invoke the find element method like below for instance
```
WebDriver decorated = new EventFiringDecorator(new WebDriverDecorator(web)).decorate(web);
        decorated.get("https://www.google.com/");
        decorated.findElement(By.name("q"));
```

On executing below exception is thrown
```
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.openqa.selenium.support.events.EventFiringDecorator.callListenerMethod(EventFiringDecorator.java:311)
	at org.openqa.selenium.support.events.EventFiringDecorator.fireAfterEvents(EventFiringDecorator.java:252)
	at org.openqa.selenium.support.events.EventFiringDecorator.lambda$afterCall$1(EventFiringDecorator.java:177)
	at java.base/java.util.Arrays$ArrayList.forEach(Arrays.java:4390)
	at org.openqa.selenium.support.events.EventFiringDecorator.afterCall(EventFiringDecorator.java:177)
	at org.openqa.selenium.support.decorators.DefaultDecorated.afterCall(DefaultDecorated.java:53)
	at org.openqa.selenium.support.decorators.WebDriverDecorator.lambda$createProxy$0(WebDriverDecorator.java:288)
	at com.sun.proxy.$Proxy6.findElement(Unknown Source)
	at decorator.EnhancedWebDriver2.main(EnhancedWebDriver2.java:27)
Caused by: java.lang.IllegalArgumentException: Argument is of an illegal type: com.sun.proxy.$Proxy7
	at org.openqa.selenium.remote.internal.WebElementToJsonConverter.apply(WebElementToJsonConverter.java:85)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.openqa.selenium.remote.RemoteWebDriver.executeScript(RemoteWebDriver.java:550)
	at decorator.WebDriverDecorator.afterFindElement(WebDriverDecorator.java:13)
	... 13 more


Process finished with exit code 0

```

## Expected behavior
Javascript executor should work even with a decorated object.

## Environment

OS: OSX
Browser: Chrome
Browser version: 90.0.4430.212
Browser Driver version: ChromeDriver 90.0.4430.24
Language Bindings version: Java 11
Selenium - 4.0.0-beta-3

**Comment by rcesarlumis** [comment]
Fixed in #9416

**Comment by diemol** [comment]
Thanks for pointing it out, @rcesarlumis. It will part of the beta 4 release.
